emptyView can accept a function that returns a view

paulfalgout opened the pull request:
In reviewing the documentation for v3's `CollectionView` this seemed to be fairly inconsistent.  It is also a moderately small change.  I'm fiddling with the `CollectionView` docs now on another branch, and if this is accepted I can make the doc updates there or it can be done in a subsequent PR.

This removes the need for `getEmptyView`

This makes they `emptyView` API match the `childView`

scott-w commented:
I'm wondering whether this could be called in a number of instances:

1. The collection initializes to empty
2. The collection is updated from populated to empty
3. When a `sync` completes - the populated was empty before and is still empty

My thinking behind 3 is to allow the `emptyView` method to return a loading spinner or "there is nothing here" message depending on whether the fetch request completes.

paulfalgout commented:
@scott-w yeah probably could. To some degree this would be possible now with `getEmptyView` in v2.  But in v3 we removed `getChildView` and replaced it with `childView` accepting a function, but didn't follow that pattern for `emptyView`  This PR is more for consistency than it is a new feature, but if there's a use-case to achieve here, we can certainly do it.. But mainly I'm focused on making v3 clean before launch more than accounting for anything new.

scott-w commented:
Fair enough, I'm just musing aloud at this point. Looks like a solid change overall.

JSteunou commented:
Good call! :+1:

denar90 commented:
👍
